### ISSUE TITLE
[JENKINS-27489] Allow GIT Publisher to work with Clone Workspace

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#2627]
<p>When using Jenkins clone workspace, we can copy the .git repository information. </p>

<p>So in scenarios like, where one jobs does the build and another does the deploy later on, we may choose to tag the source code only after it's been deployed. And if we're using Clone workspace to copy the artefacts over to the deploy job, we can't currently use GIT publisher to create tags etc. even though the .git repository is there. </p>

<p><a href="https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/hudson/plugins/git/GitPublisher.java#" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/hudson/plugins/git/GitPublisher.java#</a> Line 467<br/>
    }</p>

<p>Can we have the option to do use GitPublisher, if it detects the workspace contains a .GIT repo, and then the option to select the Credential to use appears. </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=peterlai">peterlai</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-27489" target="_blank">Allow GIT Publisher to work with Clone Workspace</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: git-plugin
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2022-11-28T01:03:22+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When using Jenkins clone workspace, we can copy the .git repository information. </p>

<p>So in scenarios like, where one jobs does the build and another does the deploy later on, we may choose to tag the source code only after it's been deployed. And if we're using Clone workspace to copy the artefacts over to the deploy job, we can't currently use GIT publisher to create tags etc. even though the .git repository is there. </p>

<p><a href="https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/hudson/plugins/git/GitPublisher.java#" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/hudson/plugins/git/GitPublisher.java#</a> Line 467
            }</p>

<p>Can we have the option to do use GitPublisher, if it detects the workspace contains a .GIT repo, and then the option to select the Credential to use appears. </p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-27489?focusedId=432719&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-432719">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Won't implement.  Publishing tags or branches should be done from a Pipeline using the <tt>withCredentials</tt> step with a git username/password credential.  That allows the user to provide credentials and perform authenticated git commands.</p>
</pre>
</details>

</ul>
</details>

<p>Won't implement.  Publishing tags or branches should be done from a Pipeline using the <tt>withCredentials</tt> step with a git username/password credential.  That allows the user to provide credentials and perform authenticated git commands.</p>
